### PR TITLE
BED-4766: API Endpoint to List OIDC Providers

### PR DESCRIPTION
**Pull request by iustinum**
## Description

This pull request introduces a unified SSO endpoint that provides detailed information on both OIDC and SAML providers. The goal is to offer a consolidated and accurate list of available authentication methods, ensuring that users are directed to the appropriate provider for successful authentication.

## Motivation and Context

This PR addresses the requirements outlined in ticket BED-4766.

Previously, SSO providers were listed using the `GET /api/v2/saml/sso endpoint`, which only supported SAML providers. With the addition of OIDC as an alternative authentication method to comply with FedRamp requirements, there is a need for a single, unified endpoint. This new endpoint will display information about both SAML and OIDC providers as a more flexible and agnostic approach. The unified endpoint will replace the existing SAML-only endpoint to ensure all authentication options are covered.

## How Has This Been Tested?

Unit Tests: Via `auth/sso_test` to verify endpoint behavior.
Integration Tests: Via `database/sso_providers_test` to verify interaction with the database.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [X] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [X] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [X] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

File: cmd/api/src/api/v2/auth/sso.go
Comment (mistahj67) [inline cmd/api/src/api/v2/auth/sso.go:137]: Couple drive by thoughts: this is making a single db hit per sso provider:
Possible option:
Since both tables are fk'd you might be able to do a more complex sql query that populates it on the fly using table joins. 

Another option: 
Snag all providers from both tables and store em in a  map[sso_id]provider, grab all sso providers then loop in the code instead and compile the response

**Review by mistahj67 — COMMENTED** [review]
I think you can point the base to @mvlipka 's create PR and it might help reviewing this without his changes as well? And then we just make sure we don't merge until his is merged. Looking good so far!

File: cmd/api/src/database/sso_providers.go
Comment (ddlees) [inline cmd/api/src/database/sso_providers.go:61]: In the same vein as what @mistahj67 had indicated earlier, you can have this function return all the sso providers and their associated provider details in one call if you use a join.

https://gorm.io/docs/query.html#Joins

**Review by ddlees — DISMISSED** [review]

File: cmd/api/src/database/sso_providers.go
Comment (mistahj67) [inline cmd/api/src/database/sso_providers.go:67]: I am generally against gorm and it'll be fun to tease this out later, but I will concede this is neat!

File: cmd/api/src/api/v2/auth/sso_test.go
Comment (mistahj67) [inline cmd/api/src/api/v2/auth/sso_test.go:70]: might help cleanup but if nothing is being mutated inside of the t.Run blocks, you can safely move all the initialization steps above the t.Run() blocks

**Review by mistahj67 — CHANGES_REQUESTED** [review]
Only blocked because the permission requirement needs to be removed. Otherwise this is looking great! I concede, the gorm joins feature is schnazzy just don't let @superlinkx know I said that out loud. 

Also did you run prep-for-codereview 👀

File: cmd/api/src/api/v2/auth/sso.go
Comment (iustinum) [reply cmd/api/src/api/v2/auth/sso.go:69]: Good point and noted!

File: cmd/api/src/database/sso_providers.go
Comment (iustinum) [reply cmd/api/src/database/sso_providers.go:67]: Yup, credit to Dillon for sending the helpful GORM doc above!

File: cmd/api/src/api/v2/auth/sso.go
Comment (mistahj67) [inline cmd/api/src/api/v2/auth/sso.go:78]: Because we leverage the if / else blocks heavily we don't need all the returns anymore

```suggestion
```

**Review by mistahj67 — APPROVED** [review]
Tested locally, everything is working solid! 🚀

**Pull request merged**
